src/test/java/jenkins/security/plugins/ldap/LDAPDNEscapingTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.security.plugins.ldap;
+
+import java.io.InputStream;
+import java.util.Collections;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.JenkinsRule.WebClient;
+import org.springframework.security.core.userdetails.UserDetails;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import hudson.security.GroupDetails;
+import hudson.security.LDAPSecurityRealm;
+import hudson.util.Secret;
+import jenkins.model.IdStrategy;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * Tests {@link LDAPConfiguration} with DNs that are not URL safe.
+ */
+@LDAPTestConfiguration
+@RunWith(JUnitParamsRunner.class)
```

Comment:
Why the special test dep, when JUnit has built-in support for parameterized tests?

Reply:
it is cleaner with this runner, you can have different tests with different parameters etc (which I had at one point) rather than littering a class with constructors or parameters.

mentioned in passing in the official junit docs for parameterized tests. https://github.com/junit-team/junit4/wiki/Parameterized-tests#see-also  so it is not that "far out"

Reply:
:shrug: just hoping to avoid a new dep.

Or you can just define a single plain test and run through these parameters in a loop, which would also be faster (there is no particular reason to start a new Jenkins service each time).

Reply:
I had various failures in the different branches so seeing a full list of the failures for the different parameters was helpful to me.